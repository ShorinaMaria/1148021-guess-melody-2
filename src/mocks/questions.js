const questions = [
  {
    type: `genre`,
    genre: `jazz`,
    answers: [
      {
        src: `http://mp3.classic-music.ru/music/vivaldi/01.mp3`,
        genre: `classic`,
      },
      {
        src: `http://dl2.mp3party.net/online/8464242.mp3`,
        genre: `pop`,
      },
      {
        src: `http://dl2.mp3party.net/online/2072.mp3`,
        genre: `jazz`,
      },
      {
        src: `http://dl2.mp3party.net/online/7625333.mp3`,
        genre: `rock`,
      },
    ],
  },
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        src: `http://mp3.classic-music.ru/music/vivaldi/01.mp3`,
        genre: `classic`,
      },
      {
        src: `http://dl2.mp3party.net/online/914916.mp3`,
        genre: `rock`,
      },
      {
        src: `http://dl2.mp3party.net/online/2072.mp3`,
        genre: `jazz`,
      },
      {
        src: `http://dl2.mp3party.net/online/8468238.mp3`,
        genre: `rock`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `Placebo`,
      src: `http://dl2.mp3party.net/online/8468238.mp3`,
    },
    answers: [
      {
        picture: `http://placehold.it/134x134`,
        artist: `John Snow`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Placebo`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `Scorpions`,
      src: `http://dl2.mp3party.net/online/7625333.mp3`,
    },
    answers: [
      {
        picture: `http://placehold.it/134x134`,
        artist: `Scorpions`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jim Beam`,
      },
    ],
  },
];

export default questions;
