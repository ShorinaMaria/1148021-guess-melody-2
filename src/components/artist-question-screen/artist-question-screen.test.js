import React from 'react';
import renderer from 'react-test-renderer';
import ArtistQuestionScreen from './artist-question-screen';
import questions from '../../mocks/questions';

it(`ArtistQuestionScreen renders correctly`, () => {
  const tree = renderer.create(
      <ArtistQuestionScreen
        question={questions[2]}
        screenIndex={1}
        onAnswer={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
