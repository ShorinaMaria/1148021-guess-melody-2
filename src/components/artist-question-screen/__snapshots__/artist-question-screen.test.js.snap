// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen renders correctly 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="timer__value"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
      onChange={[Function]}
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-0"
          name="answer"
          type="radio"
          value={0}
        />
        <label
          className="artist__name"
          htmlFor="answer-0"
        >
          <img
            alt="John Snow"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          John Snow
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          type="radio"
          value={1}
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Jack Daniels"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Jack Daniels
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-2"
          name="answer"
          type="radio"
          value={2}
        />
        <label
          className="artist__name"
          htmlFor="answer-2"
        >
          <img
            alt="Jim Beam"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Jim Beam
        </label>
      </div>
    </form>
  </section>
</section>
`;
